ui/tests/unit/adapters/ldap/role-test.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+
+module('Unit | Adapter | ldap/role', function (hooks) {
+  setupTest(hooks);
+  setupMirage(hooks);
+
+  hooks.beforeEach(function () {
+    this.store = this.owner.lookup('service:store');
+    this.adapter = this.store.adapterFor('ldap/role');
+    this.path = 'role';
+  });
+
+  test('it should make request to correct endpoints when listing records', async function (assert) {
+    assert.expect(4);
+
+    this.server.get('/ldap-test/:path', (schema, req) => {
+      assert.ok(req.queryParams.list, 'list query param sent when listing roles');
+      assert.strictEqual(
+        req.params.path,
+        this.path,
+        'GET request made to correct endpoint when listing roles'
+      );
+      return { data: { keys: ['test-role'] } };
+    });
+
+    await this.store.query('ldap/role', { backend: 'ldap-test', type: 'dynamic' });
+    this.path = 'static-role';
+    await this.store.query('ldap/role', { backend: 'ldap-test', type: 'static' });
+  });
+
+  test('it should make request to correct endpoints when querying record', async function (assert) {
+    assert.expect(2);
+
+    this.server.get('/ldap-test/:path/test-role', () => {
```

Comment:
I wonder if we should be explicit about testing the path here since it's dynamically calculated in the adapter. We could have two server.get defined, one for each expected path

Reply:
Ahh good catch I missed checking if `req.params.path === this.path` which is essentially the same as adding another request handler without duplicating the code. I'll get that updated!